src/features/klant/service-data-enricher.ts

```diff
@@ -0,0 +1,57 @@
+import { ServiceResult, type NotUndefined, type ServiceData } from "@/services";
```

Comment:
I find this service quite hard to follow and understand. Could we simplify this?

Maybe we could call about it.